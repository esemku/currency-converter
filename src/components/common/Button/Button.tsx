import clsx from 'clsx';
import React from 'react';
import useStyles from './styles';

interface IProps {
  name: string;
  disabled: boolean;
  onClick: () => void;
  className: string;
}

const Button: React.FC<IProps> = ({ name, disabled, onClick, className }) => {
  const styles = useStyles();

  /* eslint-disable react/button-has-type */
  return (
    <button
      type="button"
      className={clsx(styles.root, className)}
      disabled={disabled}
      onClick={onClick}
      data-testid="convertButton"
    >
      {name}
    </button>
  );
};

export default Button;
