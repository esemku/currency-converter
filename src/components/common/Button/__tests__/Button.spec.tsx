import { render } from '@testing-library/react';
import React from 'react';
import Button from '../Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Button
        name="name"
        disabled={false}
        onClick={jest.fn()}
        className="className"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
