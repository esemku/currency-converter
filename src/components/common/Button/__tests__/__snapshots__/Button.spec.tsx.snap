// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<div>
  <button
    class="makeStyles-root-1 className"
    data-testid="convertButton"
    type="button"
  >
    name
  </button>
</div>
`;
